jobs: attempt to backfill missing info rows on the fly

dt opened the pull request:
Mitigates many of the lingering negative effects on clusters which may have encountered #104798.

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104829)
<!-- Reviewable:end -->

dt committed:
jobs: attempt to backfill missing info rows on the fly

Mitigates many of the lingering negative effects on clusters which may
have encountered #104798.

Release note: none.
Epic: none.

adityamaru reviewed (approved):
LGTM, only nits. Maybe lets get @stevendanna to triple check

stevendanna reviewed (commented):
No additional comments from what Aditya already left.

dt commented:
TFTR!

bors r+

dt commented:
blathers backport 23.1
